spec/PublicAPI.spec.js

```diff
@@ -208,4 +212,256 @@ describe('public API supplied with invalid application id', () => {
       }
     );
   });
+
+  fdescribe('resetPassword', () => {
+    let makeRequest;
+    const re = new RegExp('^(?=.*[a-z]).{8,}');
+    let sendEmailOptions;
+    const emailAdapter = {
+      sendVerificationEmail: {},
+      sendPasswordResetEmail: options => {
+        sendEmailOptions = options;
+      },
+      sendMail: () => {},
+    };
+
+    const serverURL = 'http://localhost:8378/1';
+
+    const headers = {
+      'Content-Type': 'application/json',
+      'X-Parse-Application-Id': 'test',
+      'X-Parse-REST-API-Key': 'rest',
+      'X-Parse-Installation-Id': 'yolo',
+    };
+
+    beforeEach(() => {
+      makeRequest = reconfigureServer({
+        appName: 'coolapp',
+        publicServerURL: 'http://localhost:1337/1',
+        emailAdapter: emailAdapter,
+        passwordPolicy: {
+          validatorPattern: re,
+          doNotAllowUsername: true,
+          maxPasswordHistory: 1,
+          resetTokenValidityDuration: 0.5, // 0.5 second
+        },
+      }).then(() => {
+        const config = Config.get('test');
+        const user = new Parse.User();
+        user.setPassword('asdsweqwasas');
+        user.setUsername('test');
+        user.set('email', 'test@parse.com');
+        return user
+          .signUp(null)
+          .then(() => {
+            // build history
+            user.setPassword('aaaaaaaaaaaa');
```

Comment:
What is the purpose of setting the password here?

Reply:
hi, this is for the test `Password reset failed due to password is repeated`, it is necessary have set the user's password a second time and since maxPasswordHistory: 1, then the expected result will be a validation error, please, see comments added in the unit test about the expected error.

Thanks Manuel!
Ger